luigi/contrib/mongodb.py

```diff
@@ -212,8 +212,9 @@ def exists(self):
 
     def read(self):
         """
-        Returns the count number of the target, -1 if the collection doesn't exist
         Using the aggregate method to avoid inaccurate count if using a sharded cluster
         https://docs.mongodb.com/manual/reference/method/db.collection.count/#behavior
         """
-        return next(self.get_collection().aggregate([{'$group': {'_id': None, 'count': {'$sum': 1}}}])).get('count')
+        for res in self.get_collection().aggregate([{'$group': {'_id': None, 'count': {'$sum': 1}}}]):
+            return res.get('count', None)
+        return None
```

Comment:
Not sure this last line is necessary. If the `for` loop isn't entered, the method by default `return None`.

Reply:
Yes that's true, I just wanted to be as explicit as possible and since we use the None value for comparison in the exist() method I though it made sense to add that line but I could be wrong.

Reply:
Cool. Yeah, that's fine. Just commenting on it; not saying it was wrong.